### PR TITLE
Tehreem/openassessmentblock.save submission

### PR DESCRIPTION
**Pull request by tehreem-sadat**
**Trello Link:** _https://trello.com/c/5fLUbQdQ/23-open-response-assessment-event-openassessmentblocksavesubmission_

**Description:** _Server emits openassessmentblock.save_submission event when user save
    his response before submitting it._

**Checks before merge:**

- [ ] Reviewed
- [x] Unit Test Pass

### CODE REVIEW

File: openedx/features/caliper_tracking/tests/expected/openassessmentblock.save_submission .json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "name": "honor",
+    "type": "Person"
+  },
+  "eventTime": "2018-10-19T10:42:02.674Z",
+  "extensions": {
+    "extra_fields": {
+      "accept_language": "en;q=1.0, en;q=1.0",
+      "agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.67 Safari/537.36",
+      "asides": {},
+      "course_user_tags": {},
```
Comment (mumarkhan999) [inline openedx/features/caliper_tracking/tests/expected/openassessmentblock.save_submission .json]: `course_id` is missing.

File: openedx/features/caliper_tracking/transformers/open_response_transformers.py [hunk 2]
```diff
@@ -2,11 +2,12 @@
 Transformers for all openassessment events
 """
 from openedx.features.caliper_tracking.utils import convert_datetime
-
+import json
```
Comment (mumarkhan999) [inline openedx/features/caliper_tracking/transformers/open_response_transformers.py]: add one more blank line here.

File: openedx/features/caliper_tracking/transformers/open_response_transformers.py [hunk 3]
```diff
@@ -43,3 +45,49 @@ def openassessmentblock_peer_assess(current_event, caliper_event):
...
+
+def openassessmentblock_save_submission(current_event, caliper_event):
+    """
+    server emits openassessmentblock.save_submission event when user save
+    his response before submitting it.
+
+    :param current_event: default event log generated.
+    :param caliper_event: caliper_event log having some basic attributes.
+    :return: updated caliper_event.
+
```
Comment (mumarkhan999) [inline openedx/features/caliper_tracking/transformers/open_response_transformers.py]: remove this blank line.

File: openedx/features/caliper_tracking/transformers/open_response_transformers.py (hunk 2 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/transformers/open_response_transformers.py]: change the order of importing also. like given below

```
import json

from openedx.features.caliper_tracking.utils import convert_datetime
```

File: openedx/features/caliper_tracking/tests/expected/openassessmentblock.save_submission .json (hunk 1 above)
Comment (tehreem-sadat) [reply openedx/features/caliper_tracking/tests/expected/openassessmentblock.save_submission .json]: done

File: openedx/features/caliper_tracking/transformers/open_response_transformers.py [hunk 4]
```diff
@@ -1,12 +1,14 @@
 """
 Transformers for all openassessment events
 """
-from openedx.features.caliper_tracking.utils import convert_datetime
+import json
 
+from openedx.features.caliper_tracking.utils import convert_datetime
```
Comment (mumarkhan999) [inline openedx/features/caliper_tracking/transformers/open_response_transformers.py]: leave two blank lines before function definition.

Comment (tehreem-sadat) [reply openedx/features/caliper_tracking/transformers/open_response_transformers.py]: done

**Pull request merged**